soc/xtensa/intel_adsp/common/include/soc.h

```diff
@@ -135,4 +135,12 @@ static inline bool intel_adsp_ptr_is_sane(uint32_t sp)
 		 && sp <= CONFIG_IMR_MANIFEST_ADDR);
 }
 
+static ALWAYS_INLINE uint32_t prid(void)
+{
+	uint32_t prid;
+
+	__asm__ volatile("rsr %0, PRID" : "=r"(prid));
+	return prid;
+}
+
```

Comment:
@nashif beat you to it -> https://github.com/zephyrproject-rtos/zephyr/pull/46659